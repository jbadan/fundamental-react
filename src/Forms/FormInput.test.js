import FormInput from './FormInput';
import { mount } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

describe('<FormInput />', () => {
    const formInput = (
        <FormInput
            id='input-1'
            placeholder='Field placeholder text'
            state='warning'
            type='text' />
    );

    test('create form input item', () => {
        // create form set with form inputs
        let component = renderer.create(formInput);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    describe('Prop spreading', () => {
        test('should allow props to be spread to the FormInput component', () => {
            const element = mount(<FormInput data-sample='Sample' />);

            expect(
                element.getDOMNode().attributes['data-sample'].value
            ).toBe('Sample');
        });
    });
});
