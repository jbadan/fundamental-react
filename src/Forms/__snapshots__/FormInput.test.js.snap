// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormInput /> create form input item 1`] = `
<input
  className="fd-form__control is-warning"
  id="input-1"
  placeholder="Field placeholder text"
  type="text"
/>
`;
